### PR TITLE
Orqa F405 Pro

### PR DESCRIPTION
**Pull request by andyp1per**
Support for the Orqa F405 Pro https://enterprise.orqafpv.com/

Work sponsored by Orqa

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/hwdef.dat [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+PC11 SPI3_MISO SPI3
+
+# Chip select pins
+PB3 FLASH1_CS CS
+PB12 OSD1_CS CS
+PA4 GYRO1_CS CS
+
+# Beeper
+PA8 TIM1_CH1 TIM1 GPIO(80) ALARM
+PA8 BUZZER OUTPUT GPIO(81) LOW ALT(1)
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/hwdef.dat:49]: did not know alt config works on non UARTS!

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+The Orqa FC 3030 F405 supports OSD using OSD_TYPE 1 (MAX7456 driver) or OSD_TYPE 3 if using DJI OSD
+
+## PWM Output
+
+The Orqa FC 3030 F405 supports up to 8 PWM outputs in the JST-GH ESC and MFC connectors.
+
+The PWM is in 4 groups:
+
+ - PWM 1,4 in group1
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:47]: 1,4,5,6? in this group

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/hwdef.dat [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+define DEFAULT_SERIAL3_PROTOCOL SerialProtocol_GPS
+
+# UART5 - ESC Telemetry
+PD2 UART5_RX UART5 NODMA
+define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_ESCTelemetry
+
+# USART6 - Sik
+PC6 USART6_TX USART6 NODMA
+PC7 USART6_RX USART6 NODMA
+define DEFAULT_SERIAL6_PROTOCOL SerialProtocol_MAVLink
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/hwdef.dat:77]: should be MAVLink2

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+RC input is configured for GHST by default on the T1 (UART1_TX) pin. To support other RC protocols set SERIAL1_OPTIONS to 0. This will support all serial RC protocols except SBUS. For PPM and SBUS support on UART1_RX set BRD_ALT_CONFIG to 1.
+  
+## OSD Support
+
+The Orqa FC 3030 F405 supports OSD using OSD_TYPE 1 (MAX7456 driver) or OSD_TYPE 3 if using DJI OSD
+
+## PWM Output
+
+The Orqa FC 3030 F405 supports up to 8 PWM outputs in the JST-GH ESC and MFC connectors.
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:43]: which are bdir, and all support dshot, right?

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+The PWM is in 4 groups:
+
+ - PWM 1,4 in group1
+ - PWM 2-3 in group2
+ - PWM 5-6 in group3
+ - PWM 7-8 in group4
+
+Channels within the same group need to use the same output rate. If
+any channel in a group uses DShot then all channels in the group need
+to use DShot. Channels 1-4 support bi-directional dshot.
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:54]: what about 5,6 bdir also?

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+LiPo batteries.
+
+The correct battery setting parameters are:
+
+ - BATT_MONITOR 4
+ - BATT_VOLT_PIN 13
+ - BATT_CURR_PIN 12
+ - BATT_VOLT_MULT 8.3
+ - BATT_AMP_PERVLT 92.6
+
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:69]: should have a list of ad/gpio pins and a section on vtx pwr control...the +10v pin, right? there are four gpio output pins in pinout that I cant find defined in hwdef

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+ - SERIAL0 -> USB
+ - SERIAL1 -> UART1 (RX/GHST, DMA-enabled)
+ - SERIAL3 -> UART3 (GPS, DMA-enabled)
+ - SERIAL5 -> UART5 (ESC Telemetry)
+ - SERIAL6 -> UART6 (Sik/MAVLink)
+
+## RC Input
+
+RC input is configured for GHST by default on the T1 (UART1_TX) pin. To support other RC protocols set SERIAL1_OPTIONS to 0. This will support all serial RC protocols except SBUS. For PPM and SBUS support on UART1_RX set BRD_ALT_CONFIG to 1.
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:35]: dont you need to set half duplex also???

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md (hunk 2 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:47]: Yes

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/hwdef.dat (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/hwdef.dat:49]: Seems to work

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md (hunk 7 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:35]: It's set by default in the params

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/hwdef.dat (hunk 3 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/hwdef.dat]: Fixed

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md (hunk 4 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:43]: Just 1-4 support bi-dir, all support dshot

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md (hunk 6 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:69]: "GPIO" are actually Motor 5-8, can obviously be configured as GPIO as well. Added VTX power section.

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+RC input is configured for GHST by default on the T1 (UART1_TX) pin using half-duplex. To support other RC protocols set SERIAL1_OPTIONS to 0. This will support all serial RC protocols except SBUS. For PPM and SBUS support on UART1_RX set BRD_ALT_CONFIG to 1.
+  
+## OSD Support
+
+The Orqa FC 3030 F405 supports OSD using OSD_TYPE 1 (MAX7456 driver) or OSD_TYPE 3 if using DJI OSD
+
+## PWM Output
+
+The Orqa FC 3030 F405 supports up to 8 PWM outputs in the JST-GH ESC and MFC connectors.
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:43]:
```suggestion
The Orqa FC 3030 F405 supports up to 8 PWM outputs in the JST-GH ESC and MFC connectors. Outputs 5-8 are labeled as GPIO1-4 on the 7pin JST-GH connector
```

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+The PWM is in 3 groups:
+
+ - PWM 1,4-6 in group1
+ - PWM 2-3 in group2
+ - PWM 7-8 in group4
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:49]:
```suggestion
 - PWM 7-8 in group3
```

File: libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md (hunk 6 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/OrqaF405Pro/README.md:69]: needs to be documented...see above

**Review by Hwurzburg — COMMENTED** [review]
one more tweak

**Review by Hwurzburg — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**